optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/AbstractScoreDirector.java

```diff
@@ -172,9 +172,20 @@ public SupplyManager getSupplyManager() {
     // Complex methods
     // ************************************************************************
 
+    private void assertWorkingSolutionClassSameAsAnnotatedClass(Solution_ workingSolution) {
+        Class<Solution_> actualSolutionClass = (Class<Solution_>) Objects.requireNonNull(workingSolution).getClass();
```

Comment:
nitpick: I'd give the requireNonNull() it's own line, without capturing the return value - that reads better. I'd keep it in the setWorkingSolution() for the stacktrace if it fails and because that assert belongs there more I'd argue.